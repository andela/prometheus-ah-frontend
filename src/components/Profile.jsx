import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import routes from '../constants/routes';
import ArticleCard from './common/ArticleCard';

/**
 * Profile page
 */
class Profile extends Component {
  /**
   * render
   *
   * @return {void}
   */
  render() {
    const {
      profile, articles, followers, following, userId
    } = this.props;

    return (
      <div className="container-fluid">
        <div className="emp-profile">
          <div className="row">
            <div className="col-sm-4">
              <div className="profile-img">
                <img src={profile.image} alt="profile" height="200px" width="200px" />
              </div>
            </div>
            <div className="col-sm-8">
              <h5>
                {` ${profile.firstname}  ${profile.lastname} `}
                &nbsp;
                {
                  userId === profile.id ? (
                    <Link to={{ pathname: routes.EDIT_PROFILE_PAGE, state: { profile } }}>
                      <button className="btn btn-outline-danger btn-sm" type="button">
                        Edit Profile
                      </button>
                    </Link>) : ''
                }
              </h5>
              <p>
                <b>Email: </b>
                <span>{profile.email}</span>
              </p>
              <h6>{profile ? profile.bio : ''}</h6>
              <br />
              <p className="text-color">
                <span>{ followers.myTotalFollow ? followers.myTotalFollow : '0 ' }</span>
                Following &nbsp;&nbsp;
                <span>{ following.totalFollower ? following.totalFollower : '0 ' }</span>
                Followers
              </p>
            </div>
          </div>
          <div className="row">
            <div className="col-sm-12 profile-padding">
              <ul className="nav nav-tabs" id="myTab" role="tablist">
                <li className="nav-item">
                  <a className="nav-link active" id="home-tab" data-toggle="tab"
                    href="#home" role="tab" aria-controls="home" aria-selected="true"
                  >
                    Articles
                  </a>
                </li>
                <li className="nav-item">
                  <a className="nav-link" id="profile-tab" data-toggle="tab"
                    href="#profile" role="tab" aria-controls="profile" aria-selected="false"
                  >
                    Bookmarks
                  </a>
                </li>
              </ul>
              <div className="tab-content" id="myTabContent">
                <br />
                <div className="tab-pane fade show active" id="home"
                  role="tabpanel" aria-labelledby="home-tab"
                >
                  <div className="row">
                    {articles && articles.length > 0 ? articles.map(article => (
                      <div className="col-sm-6" key={article.id}>
                        <ArticleCard
                          article={article}
                        />
                      </div>
                    )) : 'You have not published any article'}
                  </div>
                </div>
                <div className="tab-pane fade" id="profile"
                  role="tabpanel" aria-labelledby="profile-tab"
                >
                  <h5>List of bookmarks</h5>
                  <h1>&nbsp;</h1>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Profile.propTypes = {
  profile: PropTypes.shape({
  }).isRequired,
  followers: PropTypes.shape({
  }).isRequired,
  following: PropTypes.shape({
  }).isRequired,
  userId: PropTypes.number.isRequired,
  articles: PropTypes.instanceOf(Object)
};

export default Profile;
