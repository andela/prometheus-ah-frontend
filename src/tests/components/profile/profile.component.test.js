import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Profile from '../../../components/Profile';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
describe('Profile component', () => {
  beforeEach(() => {
    const props = {
      profile: {},
      articles: [{
        body: 'Contrary to popular belief, Lorem Ipsum is not simply random text.',
        createdAt: '2018-10-02T12:07:45.830Z',
        description: 'why code in ruby',
        id: 3,
        readingTime: '3 min read',
        slug: 'how-coding-works',
        title: 'Coding',
        updatedAt: '2018-10-02T12:07:45.830Z',
        userId: 10,
      }],
      followers: {
        users: {
          message: 'You are yet to have followers.'
        }
      },
      following: {
        users: {
          message: 'You are yet to follow an Author.'
        }
      },
      userId: 10,
      bookmarks: {
        articles: [{
          body: 'Contrary to popular belief, Lorem Ipsum is not simply random text.',
          createdAt: '2018-10-02T12:07:45.830Z',
          description: 'why code in ruby',
          id: 3,
          readingTime: '3 min read',
          slug: 'how-coding-works',
          title: 'Coding',
          updatedAt: '2018-10-02T12:07:45.830Z',
          userId: 10,
        }],
        paginationMeta: {}
      }
    };
    wrapper = shallow(<Profile {...props} />);
  });

  it('should render the profile component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render articles cards and links', () => {
    expect((wrapper).find('ArticleCard').length).toBe(2);
  });
});
