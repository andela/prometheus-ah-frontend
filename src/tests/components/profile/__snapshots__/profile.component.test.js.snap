// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should render the profile component 1`] = `
<div
  className="container-fluid"
>
  <div
    className="emp-profile"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-4"
      >
        <div
          className="profile-img"
        >
          <img
            alt="profile"
            height="200px"
            src="https://image.ibb.co/i48Wqf/paceholder.jpg"
            width="200px"
          />
        </div>
      </div>
      <div
        className="col-sm-8"
      >
        <h5>
          Update Your Profile
           
        </h5>
        <p>
          <b>
            Email: 
          </b>
          <span />
        </p>
        <h6 />
        <br />
        <p
          className="text-color"
        >
          <span>
            0 
          </span>
          Following   
          <span>
            0 
          </span>
          Followers
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm-12 profile-padding"
      >
        <ul
          className="nav nav-tabs"
          id="myTab"
          role="tablist"
        >
          <li
            className="nav-item"
          >
            <a
              aria-controls="home"
              aria-selected="true"
              className="nav-link active"
              data-toggle="tab"
              href="#home"
              id="home-tab"
              role="tab"
            >
              Articles
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-controls="profile"
              aria-selected="false"
              className="nav-link"
              data-toggle="tab"
              href="#profile"
              id="profile-tab"
              role="tab"
            >
              Bookmarks
            </a>
          </li>
        </ul>
        <div
          className="tab-content"
          id="myTabContent"
        >
          <br />
          <div
            aria-labelledby="home-tab"
            className="tab-pane fade show active"
            id="home"
            role="tabpanel"
          >
            <div
              className="row"
            >
              <div
                className="col-sm-6"
                key="3"
              >
                <ArticleCard
                  article={
                    Object {
                      "body": "Contrary to popular belief, Lorem Ipsum is not simply random text.",
                      "createdAt": "2018-10-02T12:07:45.830Z",
                      "description": "why code in ruby",
                      "id": 3,
                      "readingTime": "3 min read",
                      "slug": "how-coding-works",
                      "title": "Coding",
                      "updatedAt": "2018-10-02T12:07:45.830Z",
                      "userId": 10,
                    }
                  }
                />
              </div>
            </div>
          </div>
          <div
            aria-labelledby="profile-tab"
            className="tab-pane fade"
            id="profile"
            role="tabpanel"
          >
            <div
              className="row"
            >
              <div
                className="col-sm-6"
                key="3"
              >
                <ArticleCard
                  article={
                    Object {
                      "body": "Contrary to popular belief, Lorem Ipsum is not simply random text.",
                      "createdAt": "2018-10-02T12:07:45.830Z",
                      "description": "why code in ruby",
                      "id": 3,
                      "readingTime": "3 min read",
                      "slug": "how-coding-works",
                      "title": "Coding",
                      "updatedAt": "2018-10-02T12:07:45.830Z",
                      "userId": 10,
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
